aws/resource_aws_codepipeline.go

```diff
@@ -227,8 +227,12 @@ func expandAwsCodePipelineArtifactStore(d *schema.ResourceData) *codepipeline.Ar
 
 func flattenAwsCodePipelineArtifactStore(artifactStore *codepipeline.ArtifactStore) []interface{} {
 	values := map[string]interface{}{}
-	values["type"] = *artifactStore.Type
-	values["location"] = *artifactStore.Location
+	if artifactStore.Type != nil {
```

Comment:
Performing a `nil` check before using the AWS Go SDK type conversion functions is redundant. They convert `nil` to the zero-value for the type: https://docs.aws.amazon.com/sdk-for-go/api/aws/#StringValue